deid/dicom/parser.py

```diff
@@ -451,12 +453,49 @@ def add_field(self, field, value):
         # The addition will be different depending on if we have filemeta
         is_filemeta = False
 
-        # Helper function to update dicom
+        def parse_tag_string(tag_string):
+            """
+            There are four possible varieties of `tag_string`:
+            1. A string of the form "(0002,0010)"
+            2. A string of the form "00020010"
+            3. A string of the form "2" (decimal), indicating the index within
+               a sequence
+            4. A tag name (e.g., "PatientID"), rather than a number.
+            """
+            if tag_string.startswith("("):
+                result = parentheses_hex_tag_format.match(tag_string)
+                return int(f"0x{result.group(1)}{result.group(2)}", base=16)
+            else:
+                try:
+                    if bare_hex_tag_format.match(tag_string):
+                        return int(f"0x{tag_string}", base=16)
+                    else:
+                        return int(tag_string)
```

Comment:
You can simplify this a bit - if the first statement returns, the else isn't needed.

```suggestion
                    return int(tag_string)
```

Reply:
Good catch!